### ISSUE TITLE
opam depext cannot find an already installed dependency

### ISSUE THREAD

**Issue by ulugbekna** [ocaml/opam-depext#136]
I think `opam depext <pckg>` cannot find one of the package system dependencies that are already present in the system. When I do `opam depext memtrace_viewer`, I get the following output: 

```shell
❯ opam depext memtrace_viewer

# Detecting depexts using vars: arch=x86_64, os=macos, os-distribution=homebrew, os-family=homebrew
# The following system packages are needed:
gmp
libffi
lzlib
openssl
pkg-config
Warning: Calling `brew list` to only list formulae is deprecated! Use `brew list --formula` instead.
# The following new OS packages need to be installed: openssl
```

However, `openssl` is installed: 

```shell
❯ which openssl
/usr/local/opt/openssl@1.1/bin/openssl
```

It perhaps is 
- either a misconfiguration on my part (how `opam depext` detects whether a dependency installed?) 
- or a bug in `opam depext`

---

```shell
# opam config report
# opam-version      2.0.7
# self-upgrade      no
# system            arch=x86_64 os=macos os-distribution=homebrew os-version=10.15.5
# solver            builtin-mccs+glpk
# install-criteria  -removed,-count[version-lag,request],-count[version-lag,changed],-changed
# upgrade-criteria  -removed,-count[version-lag,solution],-new
# jobs              7
# repositories      2 (http) (default repo at 34883e4b)
# pinned            0
# current-switch    /Users/ulugbekna/code/test
```

**Comment by rjbou** [comment]
opam depext uses [`brew list`](https://github.com/ocaml-opam/opam-depext/blob/2.0/depext.ml#L199), 
Can you share the output of `brew list | grep openssl` ?

**Comment by ulugbekna** [comment]
```shell
~ ❯ brew list | grep openssl
Warning: Calling `brew list` to only list formulae is deprecated! Use `brew list --formula` instead.
openssl@1.1

~ ❯ brew list --formula | grep openssl
openssl@1.1
```

**Comment by ulugbekna** [comment]
Providing some context to the issue: I'm trying to install `memtrace_viewer`, but the installation fails as described in this [issue](https://github.com/janestreet/memtrace_viewer/issues/4). I was advised to run `opam depext memtrace_viewer`, which shows that I'm missing `openssl`, so I guess that may be the reason for the installation failure.

**Comment by ulugbekna** [comment]
Could the reason for failure be that `brew list` shows `openssl@1.1` but `opam depext` is looking for just `openssl`? 

(Note that `which openssl` shows correct path to `openssl` : `/usr/local/opt/openssl@1.1/bin/openssl`)

**Comment by rjbou** [comment]
Yes, it is exactly that. It is fixed in [opam](https://github.com/rjbou/opam/commit/4c0b6654c85a436f38b191a81a25dbdf3dde4a09), but not backported into opam depext

**Comment by ulugbekna** [comment]
Great, thanks!

**Comment by rjbou** [comment]
note: it's fixed in opam 2.1 alpha/beta as external dependencies handling is embed in those versions, but 2.0.x version still need opam depext
